gno.land/cmd/gnoland/main.go

```diff
@@ -314,3 +309,103 @@ func loadGenesisBalances(path string) []string {
 	}
 	return balances
 }
+
+// listGnoPkgs lists all gno packages in the given root directory.
+func listGnoPkgs(root string) ([]pkg, error) {
+	var pkgs []pkg
+
+	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if d.IsDir() {
+			goModPath := filepath.Join(path, "gno.mod")
+			data, err := os.ReadFile(goModPath)
+			if err == nil {
+				gnoMod, err := gnomod.Parse(goModPath, data)
+				if err != nil {
+					return fmt.Errorf("parse: %w", err)
+				}
+				gnoMod.Sanitize()
+				if err := gnoMod.Validate(); err != nil {
+					return fmt.Errorf("validate: %w", err)
+				}
+				pkgs = append(pkgs, pkg{
+					name: gnoMod.Module.Mod.Path,
+					path: path,
+					requires: func() []string {
+						var reqs []string
+						for _, req := range gnoMod.Require {
+							reqs = append(reqs, req.Mod.Path)
+						}
+						return reqs
+					}(),
+				})
+				return fs.SkipDir
+			} else if !os.IsNotExist(err) {
+				return err
+			}
```

Comment:
The same here:
```suggestion
                        if err != nil && !os.IsNotExist(err) {
				return err
			}
			gnoMod, err := gnomod.Parse(goModPath, data)
			if err != nil {
				return fmt.Errorf("parse: %w", err)
			}
			gnoMod.Sanitize()
			if err := gnoMod.Validate(); err != nil {
				return fmt.Errorf("validate: %w", err)
			}
			pkgs = append(pkgs, pkg{
				name: gnoMod.Module.Mod.Path,
				path: path,
				requires: func() []string {
					var reqs []string
					for _, req := range gnoMod.Require {
						reqs = append(reqs, req.Mod.Path)
					}
					return reqs
				}(),
			})
	
			return fs.SkipDir

Reply:
This is not logically same.
```go
if err != nil && !os.IsNotExist(err) {
	return err
}
```

it will pass even if `gno.mod` not exists. And will fail when try to parse the `gno.mod` file. 

Maybe?
```go
if err != nil && !os.IsNotExist(err) {
	return err
} else {
        return nil
}
```


But I think the current code is more readable.

Reply:
Sorry for that, I did the refactor directly in the comments section.

Following [effective go](https://go.dev/doc/effective_go#control-structures), we have to avoid unnecessary else statements:

```go
if err != nil {
	return err
} 
if os.IsNotExist(err)  {
        return nil
}
```

Reply:
Should be resolved by: a3e978ad739086056b92d14ddb5f27a66df4e14c

Just reversed the order to maintain the original flow:
```go
if os.IsNotExist(err) {
	return nil
}
if err != nil {
	return err
}
```